test/e2e/csv_e2e_test.go

```diff
@@ -69,7 +69,7 @@ var _ = Describe("ClusterServiceVersion", func() {
 					Name: "test-namespace-1",
 				},
 			}
-			Expect(ctx.Ctx().Client().Create(context.Background(), &ns)).To(Succeed())
+			Eventually(Expect(ctx.Ctx().Client().Create(context.Background(), &ns)).To(Succeed()))
```

Comment:
I'm not sure how nested test specs like this behave.

You can avoid nesting by doing something like this:

```go
bctx, cancel := context.WithCancel(context.Background()) // Use a shared context to reduce boilerplate
defer cancel() // Be sure to kill any long-lived goroutines after we finish

cli := ctx.Ctx().Client()
Eventually(func() error {
     return ctx.Ctx().Client().Create(bctx, &ns)
}).Should(Succeed())
```